examples/10.dynamic_configuration/README.md

```diff
@@ -0,0 +1,125 @@
+```
+In both cases the macro can take up to 8 names, which are defines in the example to be char[16]. The difference between the macros is value they supply for the is_source element.
+
+Both methods exposed by the Config Broker require the calling compartments capability. The set_config() method also takes name and data values, and the Config Broker checks the name against the list in the unsealed capability along with the is_source value. The on_config() method takes a cheri_callback parameter, which is registers against all of the names in the capability. Although it may seem more obvious to allow a subscriber to register separate callbacks for each item (since they will have different data structures) this approach ensures that a compartment always receives all of the values it is allowed, and the name is passed to the callback to allow simple conditional logic in the callback.
+
+Using the same pattern as the allocator the Config Broker assigns an id to each capability the first time it is presented, which is used to track the association between callback function and capability in the callback lists for each item. If a compartment calls on_config() a second time then the previous callback is replaced. 
+
+## Memory Management
+In the example the ConfigSource allocates a new heap object for each updated, and frees it immediately after the call to set_config() - which is prototypical of something working from a network stack and ensured that new values do not overwrite existing ones until they have been validated.
+
+Subscribers can not be allowed to change the configuration data (it may be shared with other compartments) so the Config Broker cderives a Read Only capability to pass to subscribers.
+
+The Config Broker also need to ensure that it can provide the data to any current and future subscribers, so it adds its own claim to the object, which it only releases when it gets a new object for that item. The claim is made against the derived read only capability.  
+
+Any subscribers which need to have access the configuration data beyond the scope of the callback can also make their own claim on the object, and release it on the next update. 
+
+## Sandpit Compartment
+Although the static sealed capabilities provide protection over who can update and receive configuration items, they can not offer any assurance over the content. To treat the data as initially untrusted we have to assume that not only may it contain invalid values, but that it may be constructed so as to cause harm when it is being parsed. In the example the first action of each callback is to pass the received value to a valuation method in a sandpit container, which has no access to the callers stack or context and cannot allocate any additional heap. An error handler in the validator compartment traps any violations, and if the validator does not return a success value the compartment simply ignores the update and keeps its claim on the previous object.
```

Comment:
You probably don't need the error handler in the validator to do anything.  If it returns `int`, errors will be reported as `-ECOMPARTMENTFAIL`.